### PR TITLE
App: Add Select-2 And File-import Components to app level

### PR DESCRIPTION
**Pull request by joshmarom**
For reusability and consistency

### CODE REVIEW

**Comment by joshmarom** [comment]
@matipojo @kobizz please review asap

File: core/app/assets/js/utils/utils.js [hunk 1]
```diff
@@ -6,4 +6,15 @@ export default class Utils {
 
 		return `${ pixels * 0.0625 }rem`;
 	}
+
+	static arrayToClassName = ( array ) => {
+		return array.filter( ( classItem ) => '' !== classItem ).join( ' ' );
+	}
+
+	static stringToRemValues = ( string ) => {
```
Comment (kobizz) [inline core/app/assets/js/utils/utils.js:14]: Where this method is used?

**Review by kobizz — CHANGES_REQUESTED** [review]

**Review by kobizz — CHANGES_REQUESTED** [review]
Fix JS lint

File: core/app/assets/js/utils/utils.js (hunk 1 above)
Comment (joshmarom) [reply core/app/assets/js/utils/utils.js:14]: It's used by many components. PR will be sent soon.

File: core/app/assets/js/ui/molecules/select2-api.scss [hunk 2]
```diff
@@ -0,0 +1,151 @@
+$select-box-selected-item-default-color: #5897fb;
```
Comment (kobizz) [inline core/app/assets/js/ui/molecules/select2-api.scss:1]: Decide what to do with this color

**Review by kobizz — CHANGES_REQUESTED** [review]

**Review by kobizz — DISMISSED** [review]

File: core/app/app.php [hunk 3]
```diff
@@ -128,6 +128,13 @@ private function enqueue_assets() {
 		Plugin::$instance->init_common();
 		Plugin::$instance->common->register_scripts();
 
+		wp_register_style(
+			'select2',
+			ELEMENTOR_ASSETS_URL . 'lib/e-select2/css/e-select2.css',
```
Comment (matipojo) [inline core/app/app.php:133]: Add a minified version via `$suffix = ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG || defined( 'ELEMENTOR_TESTS' ) && ELEMENTOR_TESTS ) ? '' : '.min';`

File: core/app/app.php [hunk 4]
```diff
@@ -163,12 +171,23 @@ private function enqueue_assets() {
 			true
 		);
 
+		wp_register_script(
+			'select2',
+			ELEMENTOR_ASSETS_URL . 'lib/e-select2/js/e-select2.full.js',
```
Comment (matipojo) [inline core/app/app.php:176]: Add a minified version via `$suffix = ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG || defined( 'ELEMENTOR_TESTS' ) && ELEMENTOR_TESTS ) ? '' : '.min';`

File: core/app/assets/js/molecules/select-file.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import { useRef } from 'react';
+
+import Button from 'elementor-app/ui/molecules/button';
+
+import './select-file.scss';
+
+export default function SelectFile( props ) {
```
Comment (matipojo) [inline core/app/assets/js/molecules/select-file.js:7]: It's not a `select` field. Use `upload` or `input`.

File: core/app/assets/js/organisms/import-file.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
+import SelectFile from 'elementor-app/molecules/select-file';
+import DragDrop from 'elementor-app/ui/atoms/drag-drop';
+import Icon from 'elementor-app/ui/atoms/icon';
+import Heading from 'elementor-app/ui/atoms/heading';
+import Text from 'elementor-app/ui/atoms/text';
+
+import './import-file.scss';
+
+export default function ImportFile( props ) {
```
Comment (matipojo) [inline core/app/assets/js/organisms/import-file.js:9]: It's not an `import` field. Use `dropzone` or something like.

File: core/app/assets/js/molecules/select-file.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
+import { useRef } from 'react';
```
Comment (matipojo) [inline core/app/assets/js/molecules/select-file.js:1]: Move all `input` components to `core/app/assets/js/ui/inputs`

File: core/app/assets/js/organisms/import-file.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+	className: PropTypes.string,
+	children: PropTypes.any,
+	onFileSelect: PropTypes.func,
+	heading: PropTypes.string,
+	text: PropTypes.string,
+	secondaryText: PropTypes.string,
+	buttonText: PropTypes.string,
+	icon: PropTypes.string,
+	noIcon: PropTypes.bool,
+	noButton: PropTypes.bool,
```
Comment (matipojo) [inline core/app/assets/js/organisms/import-file.js:45]: Use the positive way. with defaults `icon=true`, `button=true`.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/app/app.php (hunk 3 above)
Comment (kobizz) [reply core/app/app.php:133]: You can use `$this->get_js_assets_url()` for this purpose

File: core/app/app.php (hunk 4 above)
Comment (kobizz) [reply core/app/app.php:176]: You can use $this->get_js_assets_url() for this purpose

File: core/app/assets/js/molecules/select-file.js (hunk 7 above)
Comment (joshmarom) [reply core/app/assets/js/molecules/select-file.js:1]: We would like to have a discussion about this issue and files architecture in general, after the release.
Let's leave it as is for now...

**Review by matipojo — DISMISSED** [review]

**Review by kobizz — CHANGES_REQUESTED** [review]

**Review by matipojo — DISMISSED** [review]

**Review by kobizz — APPROVED** [review]

File: core/app/app.php [hunk 9]
```diff
@@ -128,6 +128,13 @@ private function enqueue_assets() {
 		Plugin::$instance->init_common();
 		Plugin::$instance->common->register_scripts();
 
+		wp_register_style(
+			'select2',
+			$this->get_css_assets_url( 'e-select2','assets/lib/e-select2/css/' ),
```
Comment (kobizz) [inline core/app/app.php:133]:
```suggestion
			$this->get_css_assets_url( 'e-select2', 'assets/lib/e-select2/css/' ),
```

**Review by kobizz — CHANGES_REQUESTED** [review]

**Review by kobizz — APPROVED** [review]

**Pull request merged**